community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/helpers/simpleExpressionEvaluator.scala

```diff
@@ -38,8 +38,8 @@ object simpleExpressionEvaluator {
 
   def isNonDeterministic(expr: Expression): Boolean =
     expr.inputs.exists {
-      case (FunctionInvocation(FunctionName(Rand.name), _, _), _) => true
-      case (FunctionInvocation(FunctionName(Timestamp.name), _, _), _) => true
+      case (func@FunctionInvocation(_, _, _), _) if func.function contains Rand => true
+      case (func@FunctionInvocation(_, _, _), _) if func.function contains Timestamp => true
```

Comment:
`func.function` is an `Option[Function]`. You could do `func.function == Some(Rand)` but then intellij tells you that you should use `contains`.